Realm/RLMResults.mm

```diff
@@ -376,6 +376,9 @@ - (id)objectAtIndexedSubscript:(NSUInteger)index {
 }
 
 - (id)aggregate:(NSString *)property method:(util::Optional<Mixed> (Results::*)(size_t))method methodName:(NSString *)methodName {
+    if (_results.get_mode() == Results::Mode::Empty) {
```

Comment:
> it should just be checking for _info being null as that's the thing that actually needs to be guarded against

In a number of other functions we happen to not crash by virtue of a similar check for the results being in the `Empty` mode. It seemed cleanest to rely on the mode being `Empty` when `_info` is null, rather than checking both in all cases.